vendor/model/share_post.go

```diff
@@ -0,0 +1,38 @@
+package model
+
+import (
+	"database"
+	"entity"
+)
+
+// NewSharePost ...
+func NewSharePost(userID, postID int64) (int64, error) {
+	return database.Orm.Table("share_post").Insert(&entity.SharePost{
```

Comment:
I think the "Table" part is unnecessary, the record will be inserted into the database table which is linked with xorm type of SharePost.